README.md

```diff
@@ -237,19 +236,32 @@ without the agent. In order to do that, the `provisionVMAgent` flag must be set
 provisioning time, via whichever API is being used. We will provide more details on
 this on our wiki when it is generally available. 
 
-#### __Provisioning.Enabled__
+#### __Provisioning.Agent__
 
-_Type: Boolean_  
+_Type: String_
+_Default: auto_
+
+Choose which provisioning agent to use (or allow waagent to figure it out by
+specifying "auto"). Possible options are "auto" (default), "waagent", "cloud-init",
+or "disabled".
+
+#### __Provisioning.Enabled__ (*removed in VERSION*)
```

Comment:
wouldn't removing this be a breaking change? who are the likely users of those settings? thanks

Reply:
These settings need to be made in the image from which a VM is created, and they're generally set by a user only when they create a new image. Simply updating waagent in an already-provisioned VM should have no impact, as the settings are ignored (since provisioning already happened).

I can imagine someone being impacted if they take an image they'd already created, update the waagent, and use that to create a new image. It might be worthwhile for the waagent package (deb or rpm) to have a post-install script check the config file for the presence of the `Provisioning.Enabled` config setting and emit a warning or error.

Reply:
Might also be a good idea for the provisioning code in waagent to detect the presence of `Provisioning.Enabled` and emit a warning in the log (and via telemetry). This should make diagnosis easier in the few cases where the change gets missed.